src/qutip_cupy/dense.py

```diff
@@ -0,0 +1,130 @@
+"""" This module contains the ```CuPyDense``` class as well as all its 
+asociated function conversion and specializations functions as 
+required by QuTiP's Data Layer """
+
+import numbers
+
+import cupy as cp
+from qutip.core import data
+
+
+class CuPyDense(data.Data):
+    def __init__(self, data, shape=None, copy=True, dtype=cp.complex128):
+        """ 
```

Comment:
This docstring for ``__init__`` should be on the class itself (this is convention, but also makes a lot of sense since it usually also contains documentation on how to use the class). E.g.
```python
class CuPyDense(data.Data):
    """
    This class provides a dense matrix ...
    """

    def __init__(self, data, ...):
        ...
```

Reply:
423ac6a